libclc/amdgcn-amdhsa/libspirv/misc/sub_group_shuffle.cl

```diff
@@ -8,16 +8,15 @@
 
 #include <spirv/spirv.h>
 
-#define SELF __builtin_amdgcn_mbcnt_hi(-1, __builtin_amdgcn_mbcnt_lo(-1, 0))
+#define SELF __spirv_SubgroupLocalInvocationId();
 #define SUBGROUP_SIZE __spirv_SubgroupMaxSize()
 
 // Shuffle
 // int __spirv_SubgroupShuffleINTEL<int>(int, unsigned int)
 _CLC_DEF int
 _Z28__spirv_SubgroupShuffleINTELIiET_S0_j(int Data, unsigned int InvocationId) {
-  int self = SELF;
```

Comment:
Could you please explain this change?

Reply:
When you do `self & ~(SUBGROUP_SIZE - 1)` you are masking out the `self` val (unless it is greater than `(SUBGROUP_SIZE - 1)` in which case you get the bits which are greater. In all cases the calculation for `__builtin_amdgcn_mbcnt_hi(-1, __builtin_amdgcn_mbcnt_lo(-1, 0))` should return something smaller than `SUBGROUP_SIZE - 1` so this masking was just adding zero to `InvocationId` in a fancy way.

Meaning what was really happening previously was:

```
_CLC_DEF int
_Z28__spirv_SubgroupShuffleINTELIiET_S0_j(int Data, unsigned int InvocationId) {
   return __builtin_amdgcn_ds_bpermute(InvocationId << 2, Data);
}
```

This way is in fact probably preferable to doing the current call to `__spirv_SubgroupLocalInvocationId`